tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -117,32 +144,29 @@ ZTEST_USER(uart_async_single_read, test_single_read)
 	zassert_equal(k_sem_take(&rx_buf_released, K_MSEC(100)),
 		      0,
 		      "RX_BUF_RELEASED timeout");
+	uart_rx_disable(uart_dev);
+
 	zassert_equal(k_sem_take(&rx_disabled, K_MSEC(1000)), 0,
 		      "RX_DISABLED timeout");
 	zassert_equal(k_sem_take(&rx_rdy, K_MSEC(100)), -EAGAIN,
 		      "Extra RX_RDY received");
 
-	zassert_equal(memcmp(tx_buf, rx_buf+5, 5), 0, "Buffers not equal");
-	zassert_equal(tx_aborted_count, 0, "TX aborted triggered");
+	zassert_equal(memcmp(tx_buf, tdata.last_rx_buf, 5), 0, "Buffers not equal");
+	zassert_equal(tdata.tx_aborted_count, 0, "TX aborted triggered");
 }
 
 static void *multiple_rx_enable_setup(void)
 {
 	uart_async_test_init();
 
-	tx_aborted_count = 0;
-
+	memset(&tdata, 0, sizeof(tdata));
+	tdata.rx_buf_req_done = true;
```

Comment:
I think it would be better to rename `rx_buf_req_done` to e.g. `supply_next_buffer` and set it to `true` only in `test_single_read()` to better reflect the purpose of this member.